crates/jstz_core/src/kv/transaction.rs

```diff
@@ -55,6 +62,8 @@ pub struct Transaction {
     // A stack of transactional snapshots
     stack: Vec<Snapshot>,
     lookup_map: LookupMap,
+    rollup_outbox_queue: RollupOutboxQueue,
+    outbox_queue_meta: OutboxQueueMeta,
 }
```

Comment:
This still leaks the invariant slightly when using `Transaction::default` since you can get a `len` of `0` when `OUTBOX_QUEUE` might not actually have a length of `0`. I know that `Transaction::load` solves this, but perhaps we should remove `Transaction::default`. 

Alternatively, we split the meta into the non-persistent length and the persistent length and lazily load the persistent length when required and maintain it? Non-persistent length could be maintained as a field in `Transaction` and persistent length managed by `RollupOutboxQueue`?